android/src/main/java/com/ryanheise/audioservice/AudioService.java

```diff
@@ -202,7 +203,16 @@ void setState(List<NotificationCompat.Action> actions, int actionBits, int[] com
 				.setState(playbackState, position, speed, updateTime);
 		mediaSession.setPlaybackState(stateBuilder.build());
 
-		updateNotification();
+		if(playbackState==PlaybackStateCompat.STATE_PAUSED) stopForegroundOnPause();
+		if (androidStopForegroundOnPause && !isForeground && playbackState == PlaybackStateCompat.STATE_PLAYING)
```

Comment:
I think entering `STATE_PAUSED` is straightforward enough to detect, but the inverse of `STATE_PAUSED` is not `STATE_PLAYING` but maybe a union of various states. Maybe this:

```java
private boolean isForegroundState(int playbackState) {
    switch (playbackState) {
        case PlaybackStateCompat.STATE_NONE:
        case PlaybackStateCompat.STATE_STOPPED:
        case PlaybackStateCompat.STATE_PAUSED:
        case PlaybackStateCompat.STATE_ERROR:
            return false;
        default:
            return true;
    }
}
```

This would handle the case where on resuming after a pause, it immediately hits the buffering state.

Reply:
LGTM

Reply:
You can have a look at 81d0d14 commit.

Reply:
This would keep the service in foreground when audio is buffering for too long & thus will not get killed by the OS.

Reply:
The issue with the current commit is that when we transition to foreground state we are not setting `isForeground`. I am now thinking that we can call play even in buffering state I guess. I will update the commit.